http/transport-component.md

```diff
@@ -0,0 +1,121 @@
+# HTTP Transport Component <!-- omit in toc -->
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+| --------------- | ------------- | ------ | --------------- |
+| 1A              | Working Draft | Active | r0, 2023-05-31  |
+
+Authors: [@marcopolo]
+
+Interest Group: [@marcopolo], [@mxinden], [@marten-seemann]
+
+[@marcopolo]: https://github.com/marcopolo
+[@mxinden]: https://github.com/mxinden
+[@marten-seemann]: https://github.com/marten-seemann
+
+## Table of Contents <!-- omit in toc -->
+- [Context](#context)
+- [What is an HTTP transport](#what-is-an-http-transport)
+- [Multiaddr representation](#multiaddr-representation)
+- [HTTP Paths](#http-paths)
+
+
+## Context
+
+This document is only about advertising support for an HTTP transport. It
+doesn't make any assertions about how libp2p should interact with that
+transport. That will be defined in a future document.
+
+This exists to clarify the role of the `/http` component in Multiaddrs early to
+avoid confusion and conflicting interpretations.
+
+## What is an HTTP transport
+
+An HTTP transport is simply a node that can speak some standardized version of
+HTTP that is at least HTTP/1.1. Currently that means HTTP/1.1, HTTP/2, and
+HTTP/3. Intuitively if you can `curl` it with HTTP, then it speaks HTTP.
+
+Most environments will have a way to create an HTTP Client and Server, and the
+specific HTTP version used will be opaque. The client will negotiate (or [learn
+about](https://www.rfc-editor.org/rfc/rfc9114.html#section-3.1.1)) the specific HTTP version to use when communicating
+with the HTTP server.
+
+For this opaque case, we use the `/http` component at the end of the multidadr.
+If a node wants to advertise specific HTTP versions it may advertise a
+`/http-1.1`, `/http-2`, or `/h3` instead. Although these components are not
+currently [registered](https://github.com/multiformats/multiaddr), this document
+claims their use.
+
+
+## Multiaddr representation
+
+The multiaddr of a node with an HTTP transport ends with `/http` and is prefixed
+by information that would let an HTTP client know how to reach the server
+(remember that multiaddrs are [interpreted right to
+left](https://github.com/multiformats/multiaddr#interpreting-multiaddrs)). 
+
+The following are examples of multiaddrs for HTTP transport capable nodes:
+
+* `/dns/example.com/tls/http`
+* `/dns/example.com/https`
```

Comment:
Missing IP protocol and port.

Reply:
Is there any precedent for this? Multiaddrs are all about being explicit.

Reply:
Yes, off the top of my head: https://github.com/multiformats/js-multiaddr-to-uri#usage converts `('/dnsaddr/protocol.ai/https') -> https://protocol.ai`.

This is in line with how to interpret multiaddrs. The stuff left of the `/http` component goes to the HTTP transport. As long as the HTTP transport knows how to handle `/dns/example.com` that's all that's necessary.

Another example is the `webrtc` transport. We don't specify SCTP anywhere in there.

Reply:
That’s because there’s no WebRTC without SCTP, as there also is no QUIC v1 without TLS.

Reply:
We also don't specify if the webrtc connection is over UDP vs TCP (since I believe you can run it on top of TCP as well). This is handled automatically by the webrtc transport. There's actually a lot we don't specify in the `/webrtc` transport. The only thing that is specified is the bootstrap connection.

But anyways, my earlier point stands. Anything that is valid for an http transport is valid for the components left of `/http`.

Reply:
Actually, I agree. If you're doing DNS, you have SVCB, so that's fine.